### ISSUE TITLE
Document metric support for lua transform

### ISSUE THREAD

**Issue by jszwedko** [timberio/vector#6186]
Metrics can be accessed and manipulated via `event.metric` in a lua transform (including publishing metrics for log inputs), but this isn't documented.

We should expand the lua transform documentation to show metric support including examples.

**Comment by bruceg** [comment]
This raises a question for the metric data. In #6034 I took pains to maintain compatibility between the new metric distribution/summary/etc data model and the existing lua exposing of that data. If that isn't actually documented (and hence not used) would it make sense to drop the compatibility and document the new format going forward instead of the old one.

**Comment by binarylogic** [comment]
Yes, we will document the new format. The likelihood of breaking lua scripts with your changes is very low.

**Comment by bruceg** [comment]
To be clear, the changes in #6034 did not change the format that is exposed to Lua scripts. Instead, the Lua script wrapper reshapes the metric data as it goes in or out. If we want to proceed with the new format with no array reshaping, I'll submit another issue and PR to change scripts wrapper to the new format.
